src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import * as Yup from 'yup';
+import {screen} from '@testing-library/react';
+import SQFormDropdown from '../SQFormDropdown';
+import {renderSQForm} from '../../../utils/tests';
+import userEvent from '@testing-library/user-event';
+
+const DEFAULT_PROPS = {
+  name: 'dropdownExample',
+  label: 'Dropdown Example'
+};
+
+const DEFAULT_INTITIAL_VALUES = {
+  initialValues: {
+    dropdownExample: ''
+  }
+};
+
+const DEFAULT_OPTIONS = [
+  {label: 'Test 1', value: 1},
+  {label: 'Test 2', value: 2},
+  {label: 'Test 3', value: 3}
+];
+
+const createDropdown = (
+  options = DEFAULT_OPTIONS,
+  fieldProps = DEFAULT_PROPS
+) => {
+  return <SQFormDropdown {...fieldProps}>{options}</SQFormDropdown>;
+};
+
+it('should render list of options', () => {
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList).toBeInTheDocument();
+  expect(optionsList.children.length).toBe(3);
+});
+
+it('should render with initial value', () => {
+  const initialValues = {
+    dropdownExample: 2
+  };
+
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, {initialValues});
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 2');
+});
+
+it('should update when option is selected', () => {
+  const handleChange = jest.fn();
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    onChange: handleChange
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const option = screen.getByText('Test 1');
+  userEvent.click(option);
+
+  expect(handleChange).toHaveBeenCalled();
+  expect(handleChange).toHaveBeenCalledTimes(1);
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 1');
+});
+
+it('should disable option', () => {
+  const options = [
+    {label: 'Test 1', value: 1},
+    {label: 'Test 2', value: 2, isDisabled: true},
+    {label: 'Test 3', value: 3}
+  ];
+
+  const renderDropdown = createDropdown(options, DEFAULT_PROPS);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList.childNodes[1]).toHaveAttribute('aria-disabled', 'true');
```

Comment:
Question - How do we know what to check for to ensure the child node is disabled and that it gets properly disabled?

An `aria` property ensures accessibility.
A CSS class should appear to ensure it's styled properly
An HTML property `disabled` should appear to ensure it cannot be clicked on.

Which of these is preferred to test against to ensure the element is actually being disabled?
I'm learning on the fly so I apologize if this question is a stupid one.

Reply:
I think this is an issue with Mui. Seems like they just add `aria-disabled` and a `Mui-disabled` class, no `disabled` attribute.
Maybe testing that when clicked it doesn't reveal the options would suffice?

Reply:
> Maybe testing that when clicked it doesn't reveal the options would suffice?

What do you have in mind for how to write this test? Essentially, when clicking the disabled option, we expect nothing to happen.

Reply:
> What do you have in mind for how to write this test? Essentially, when clicking the disabled option, we expect nothing to happen.

My though process is:
- expect options list `.not.toBeInTheDocument` or `.not.toBeVisible`
- click disabled input
- expect options list still `.not.toBeInTheDocument` or `.not.toBeVisible`

That sounds clunky, but it does match how the user should experience it. Though there may be a better way to write that